### ISSUE TITLE
Optional fields not enabled after calling setValue()

### ISSUE THREAD

**Issue by sweetgiorni** [json-editor/json-editor#917]
### General information

- json-editor version:

https://cdn.jsdelivr.net/npm/@json-editor/json-editor@latest/dist/css/jsoneditor.min.css
https://cdn.jsdelivr.net/npm/@json-editor/json-editor@latest/dist/jsoneditor.js
https://cdn.jsdelivr.net/npm/bootstrap@3.4.0/dist/css/bootstrap.min.css
https://cdn.jsdelivr.net/npm/bootstrap@3.4.0/dist/js/bootstrap.min.js
https://cdn.jsdelivr.net/npm/dompurify@latest/dist/purify.min.js
https://cdn.jsdelivr.net/npm/handlebars@latest/dist/handlebars.min.js
https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js

### Expected behavior

When the show_opt_in option is 'true', setting the value with setValue() should check the opt_in tickbox automatically for all optional fields that are present in the input value.

### Actual behavior
The opt_in box remains unchecked and the form disabled even though the optional value is present in the input JSON.

### Steps to reproduce the behavior

https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgAJEtXqUIZVCgQUAelMda8ALQ61ZAHTSA5qYAmUskSjWADAHZTO1kAYgUoQQgqQjBObnE2WChsaP0AUQAPMjoIFIACAGUVewUpKgBHERgyt1QAbUV6Kl4oKgy+IhgqbFqAXTYSGTJxfQxhqmdpLuQ2PDJETABhccmpQQAVGGTUkEX6OgQ8gDEunpmQCClIKilYGg1Ghha2jtPatBAIqMJEKCkYeDOcJbFKEApNPKtdp5TrdWoAXzYkFgCDkLTIBA+Xx2sS4PGB20IAHloHB4HI8usMaUKlUavVFFEMDA0QA3OQiaL9C5XKK3aYPRBMlnYXjs7Cch7YwgA1rOG4E0H6ACCeQAajdBIVhRSAHIiOh4BVsNxUIhkETYEa+eG221sRDKMAUXg3K5SZD6WU3Mg8MnhXQ7PBgMBQX6eCAAFhplWqVDcvDwgl4pvNlpG5uwiCoJpgQoEybKsZqvBwglQmezDqdLuRvABqD+nLYkl4ZDcbi2ZLRl2u/PuKCbOZAZA4ZAyFbkVZAncQGJSrs7fEsCEbUmbM7z8+aWGwAgg2bXG9n294vb5d09Q9zc7wC7IUk8yfb70rw5Pd+aD6fvDK0lNUiTlm75bp+baPkIKbdFQrRAdOVDkmB36QVgEDligb5sMhyZYPAfxgFmLSQHBIG3veEHJqaKStG2e6/s6nqYZuZFfhRUHUc0Ai/PRFAkbaQA===

That link starts with the optional tab named "optional_tab" visible, but unchecked. It is not present in the value:
```javascript
{
  "some_text_field": ""
}
```

Checking the opt_in box for "optional_tab" changes the value to this, which is correct:

```javascript
{
  "some_text_field": "",
  "optional_tab": {
    "special_value": 0
  }
}
``` 

Now refresh the page and try enabling the optional_tab only by setting the value - not by checking the opt_in box. Set it to this:

```javascript
{
  "some_text_field": "",
  "optional_tab": {
    "special_value": 15
  }
}
``` 

Click "Update Form" and open the "Optional Tab" tab. You will see that although "optional_tab" is present in the input value and "special_value" is set to 15, the tab is unchecked:
![image](https://user-images.githubusercontent.com/12527888/107862337-14ec8680-6e01-11eb-9059-e50aeae9285e.png)

I also found that after doing this, the result of getValue() actually removes the "optional_tab" dict entirely.

Also, when I first ran into this issue I was having issues with the optional tabs not being present at all after calling setValue(), but I haven't been able to recreate it in the playground. I used the workaround described in a similar issue #610:
```javascript
editor.editors.root.setValue(initialData, true)
```
That fixed the issue with the optional tabs not being present at all, but they are still unchecked. 

Anyway, this is the workaround I made just to get things delivered on time :)

```javascript
// This is a workaround for a bug in json-editor
// If you set the value of an optional property when the opt-in option is enabled,
// The checkbox won't automatically enable. So, we do it manually
const EnableEditors = (obj, object_path) => {
  for (const key of Object.keys(obj)) {
    const subPath = object_path + "." + key;
    const editor = this.editor.getEditor(subPath);
    if (editor) {
      try {
        editor.activate()
      } catch(err) {
        // Don't care, the activate function is buggy
      }
    }
    if (typeof obj[key] === 'object') {
      EnableEditors(obj[key], subPath);
    }
  }
}

EnableEditors(val, "root");
```

**Comment by phispi** [comment]
Could be a duplicate of #807

**Comment by germanbisurgi** [comment]
fixed in #943
